app/models/odk/response_parser.rb

```diff
@@ -0,0 +1,279 @@
+        Option.where(id: option_node_str).pluck(:id).first
+      end
+    end
+
+    def populate_multimedia_answer(answer, pending_file_name, question_type)
+      if files[pending_file_name].present?
+        answer.pending_file_name = nil
+        case question_type
+        when "image", "annotated_image", "sketch", "signature"
+          answer.media_object = Media::Image.create(item: files[pending_file_name])
+        when "audio"
+          answer.media_object = Media::Audio.create(item: files[pending_file_name])
+        when "video"
+          answer.media_object = Media::Video.create(item: files[pending_file_name])
+        end
+      else
+        answer.value = nil
+        answer.pending_file_name = pending_file_name
+      end
+      answer
+    end
+
+    def populate_answer_value(answer, content, form_item)
+      question_type = form_item.qtype.name
+      return populate_multimedia_answer(answer, content, question_type) if form_item.qtype.multimedia?
+
+      case question_type
+      when "select_one"
+        answer.option_id = option_id_for_submission(content)
+      when "select_multiple"
+        unless content == "none"
+          content.split(" ").each do |oid|
+            answer.choices.build(option_id: option_id_for_submission(oid))
+          end
+        end
+      when "date", "datetime", "time"
+        # Time answers arrive with timezone info (e.g. 18:30:00.000-04), but we treat a time question
+        # as having no timezone, useful for things like 'what time of day does the doctor usually arrive'
+        # as opposed to 'what exact date/time did the doctor last arrive'.
+        # If the latter information is desired, a datetime question should be used.
+        # Also, since Rails treats time data as always on 2000-01-01, using the timezone
+        # information could lead to DST issues. So we discard the timezone information for time questions only
+        # We also make sure elsewhere in the app to not tz-shift time answers when we display them.
+        # (Rails by default keeps time columns as UTC and does not shift them to the system's timezone.)
+        content = content.gsub(/(Z|[+\-]\d+(:\d+)?)$/, "") << " UTC" if answer.qtype.name == "time"
+        answer.send("#{answer.qtype.name}_value=", Time.zone.parse(content))
+      else
+        answer.value = content
+      end
+      answer
+    end
+
+    def form_item(name)
+      form_item_id = form_item_id_from_tag(name)
+      unless FormItem.exists?(form_item_id)
+        raise SubmissionError, "Submission contains unidentifiable group or question."
+      end
+      form_item = FormItem.find(form_item_id)
+      unless form_item.form_id == response.form.id
+        raise SubmissionError, "Submission contains group or question not found in form."
+      end
+      form_item
+    end
+
+    # TODO: refactor mapping to one shared place accessible here and from odk decorators
+    def form_item_id_from_tag(tag)
```

Comment:
I think you can simplify this one a lot. Try: `tag.match(/(qing|grp)(.+?)(_\d+)?/) { |m| m[2] }`

Reply:
Yes, this has moved, but I would like your opinion on the current style (issue 8602)